android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -72,11 +81,31 @@ public CodePush(String deploymentKey, Context context, boolean isDebugMode) {
         initializeUpdateAfterRestart();
     }
 
-    public CodePush(String deploymentKey, Context context, boolean isDebugMode, String serverUrl) {
+    public CodePush(String deploymentKey, Context context, boolean isDebugMode, @NonNull String serverUrl) {
         this(deploymentKey, context, isDebugMode);
         mServerUrl = serverUrl;
     }
 
+    public CodePush(String deploymentKey, Context context, boolean isDebugMode, @NonNull String serverUrl, Integer publicKeyResourceDescriptor) {
+        this(deploymentKey, context, isDebugMode, serverUrl);
+
+        if (publicKeyResourceDescriptor != null) {
+            try {
+                mPublicKey = mContext.getString(publicKeyResourceDescriptor);
+            } catch (Resources.NotFoundException e) {
+                throw new CodePushUnknownException(
```

Comment:
I see that we have several places where we throw an `UnknownException` but why don't we add a more meaningful name to the exception class e.g. something like `CodePushInvalidPublicKeyException`? 
The key point here from my side here - is this a really Unknown exception or do we know what kind of  exception it is?